packages/js/product-editor/src/components/number-control/number-control.tsx

```diff
@@ -54,13 +55,14 @@ export const NumberControl: React.FC< NumberProps > = ( {
 	help,
 	error,
 	onBlur,
+	onFocus,
 	required,
 	tooltip,
 	placeholder,
 	disabled,
 	step = 1,
-	min = -Infinity,
-	max = +Infinity,
+	min = -1000000000000000,
+	max = 1000000000000000,
```

Comment:
I think we should keep this as `Infinity`.
Is there no better way to prevent the parsing to `1e+10...`?
Also for the download limit example, I think it does make sense to add a reasonable `max`, especially for the expiry as that is measured in `days`.

Reply:
If we wanted we could add the above max to the fields individually.